nullifier-merkle-tree/src/lib.rs

```diff
@@ -0,0 +1,481 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum NullifierMerkleTreeError {
+    #[error("Low nullifier not found")]
+    LowNullifierNotFound,
+}
+
+#[derive(Default, Debug, PartialEq)]
+pub struct Node {
+    value: [u8; 32],
+    next_index: usize,
+    next_value: [u8; 32],
```

Comment:
Do we need this value?
Since if we store the next index we can just get the value from that node.
I think we just need to handle the edge case of the highest value, thus that there is no next value. For that we could just set it to zero.

Reply:
We don't, I removed it.

Also, the case of the highest value is not that relevant. For highest value, `next_index` is 0. And the node 0 is always going to have value 0.

I think the only reason this field is included in Aztec's specification is being used in the hash.